rust/src/measures/mod.rs

```diff
@@ -67,13 +67,83 @@ impl Clone for PrivacyProfile {
 }
 
 impl PrivacyProfile {
-    pub fn new(epsilon: impl Fn(&f64) -> Fallible<f64> + 'static + Send + Sync) -> Self {
-        PrivacyProfile(Arc::new(epsilon))
+    pub fn new(delta: impl Fn(f64) -> Fallible<f64> + 'static + Send + Sync) -> Self {
+        PrivacyProfile(Arc::new(delta))
     }
 
-    // these functions allow direct invocation as a method, making parens unnecessary
-    pub fn epsilon(&self, delta: &f64) -> Fallible<f64> {
-        (self.0)(delta)
+    pub fn epsilon(&self, delta: f64) -> Fallible<f64> {
+        // reject negative zero
+        if delta.is_sign_negative() {
+            return fallible!(FailedMap, "delta must not be negative");
+        }
+
+        if !(0.0..=1.0).contains(&delta) {
+            return fallible!(FailedMap, "delta must be between zero and one");
+        }
```

Comment:
The first check seems to be covered by the second. Also, do we want to include the actual value of delta in the message? Last week there was a PR that added a lot of values to messages to help with debugging, and I'd think that would be the default going forward.

Reply:
It does two things: rejects negative zero, and gives a better error message.